### ISSUE TITLE
Multiple dispatch for operators

### ISSUE THREAD

**Issue by eitsupi** [pola-rs/r-polars#1274]
If multiple dispatches are to be achieved with S3 class methods, an approach such as the `vctrs` package may be necessary.

**Comment by eitsupi** [comment]
Given that some packages are starting to adopt S7, is it worth adopting S7 rather than implementing our own multiple-dispatch thingy? (although of course we need to prioritize a rewrite based on savvy and rlang first)

**Comment by etiennebacher** [comment]
> although of course we need to prioritize a rewrite based on savvy and rlang first

I agree with this, there are quite many things left to do in the rewrite so let's not add more complexity with this. It will be easier to weigh pros and cons once this big effort is finished.

But I'm interested in S7, I also need to learn more about it.

**Comment by eitsupi** [comment]
Yes, of course, applying S7 right now would complicate things and should be avoided.

But given that the multiple dispatch of base R operators (`+`, etc.) seems quite incomplete, the adoption of S7 seems to be an advantage.
In my opinion, this feature is not that important and it may be better to hold off on implementing it until S7 is ready to be adopted.
